sound/soc/sof/core.c

```diff
@@ -361,9 +362,10 @@ static int sof_probe(struct platform_device *pdev)
 
 	return 0;
 
-comp_err:
+mach_err:
 	snd_soc_unregister_component(&pdev->dev);
-	snd_sof_free_topology(sdev);
+comp_err:
+	snd_sof_free_topology(sdev); // FIXME: is this needed
```

Comment:
@plbossart I dont think topology is loaded at this point. So we dont need to free it yet.

Reply:
I removed the error handling from the update.